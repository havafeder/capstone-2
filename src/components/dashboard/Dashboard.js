import React, { Fragment, useState, useEffect } from 'react';

//components
import InputTodo from './todolist/InputTodo';
import ListTodos from './todolist/ListTodos';

const Dashboard = ({ setAuth }) => {

	const [name, setName] = useState('');
	const [allTodos, setAllTodos] = useState([]);
	const [todosChange, setTodosChange] = useState(false);

	async function getName() {
		try {
			const response = await fetch('http://localhost:5000/dashboard/', {
				method: 'GET',
				headers: { token: localStorage.token }
			});

			const parseRes = await response.json();
			setAllTodos(parseRes);
			setName(parseRes[0].username);
		} catch (err) {
			console.error(err.message);
		}
	}

	const logout = (e) => {
		e.preventDefault();
		localStorage.removeItem('token');
		setAuth(false);
	};

	useEffect(() => {
		getName();
		setTodosChange(false);
	}, [todosChange]);

	return (
		<Fragment>
			<div className='d-flex mt-5 justify-content-around'>
				<h1>{name.toLowerCase()}, it's time to do the damn thing.</h1>
				<button className='btn btn-primary'
					onClick={e => logout(e)}>logout</button>
			</div>
			<InputTodo setTodosChange={setTodosChange}/>
			<ListTodos allTodos={allTodos} setTodosChange={setTodosChange} />
		</Fragment>
	)
};

export default Dashboard;