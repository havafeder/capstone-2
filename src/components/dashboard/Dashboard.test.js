import React from "react";
import { render } from "@testing-library/react";
import Dashboard from "./Dashboard";
import { MemoryRouter } from "react-router";

it("matches snapshot 1", function () {
	const { asFragment } = render(
		<MemoryRouter>
			<Dashboard />
		</MemoryRouter>,
	);
	expect(asFragment()).toMatchSnapshot();
});