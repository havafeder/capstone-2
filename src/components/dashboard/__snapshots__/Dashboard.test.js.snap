// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex mt-5 justify-content-around"
  >
    <h1>
      hi , here is your to-do list.
    </h1>
    <button
      class="btn btn-primary my-5 logout"
    >
      logout
    </button>
  </div>
  <form
    class="d-flex"
  >
    <input
      class="form-control input"
      placeholder="add todo"
      type="text"
      value=""
    />
    <button
      class="btn add"
    >
      add
    </button>
  </form>
  <table
    class="table mt-5"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          to-do:
        </th>
        <th
          scope="col"
        >
          edit
        </th>
        <th
          scope="col"
        >
          delete
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</DocumentFragment>
`;

exports[`matches snapshot 1 1`] = `
<DocumentFragment>
  <div
    class="my-5"
  >
    <div />
  </div>
  <div
    class="d-flex mt-5 justify-content-around"
  >
    <h1>
      hi , here is your to-do list.
    </h1>
    <button
      class="btn btn-primary my-5 logout"
    >
      logout
    </button>
  </div>
  <form
    class="d-flex"
  >
    <input
      class="form-control input"
      placeholder="add todo"
      type="text"
      value=""
    />
    <button
      class="btn add"
    >
      add
    </button>
  </form>
  <table
    class="table mt-5"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          to-do:
        </th>
        <th
          scope="col"
        >
          edit
        </th>
        <th
          scope="col"
        >
          delete
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</DocumentFragment>
`;
